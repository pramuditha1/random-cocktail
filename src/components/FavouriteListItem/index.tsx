import * as React from "react";
import Card from "@mui/material/Card";
import CardMedia from "@mui/material/CardMedia";
import Lable from "../Lable";
import ButtonComp from "../Button";
import AddIcon from "@mui/icons-material/Add";
import RemoveIcon from "@mui/icons-material/Remove";
export interface FavouriteListItemPropTypes {
  name: string;
  quantity: number;
  description?: string;
  image: string;
  addItem: () => void;
  removeItem: () => void;
  className?: string;
}

const FavouriteListItem: React.FC<FavouriteListItemPropTypes> = ({
  name,
  quantity = 1,
  image,
  addItem,
  removeItem,
  className,
}) => {
  return (
    <Card
      className={`flex items-center border border-gray-200 p-3 space-x-4  overflow-y-auto ${className}`}
      sx={{ minWdth: 500, minHeight: 100, margin: 0.5 }}
    >
      <CardMedia
        component="img"
        alt={`${name}-image`}
        className="w-24 h-24 object-cover rounded"
        image={image}
      />
      <div className="flex flex-col items-center w-full space-y-1">
        <Lable variant="h6" text={name} styles={{ justifyContent: "center" }} />
        <div className="flex items-center space-x-2">
          <ButtonComp
            variant="outlined"
            startIcon={<AddIcon />}
            onClickHandler={addItem}
            style={{ padding: "5px 1px 5px 10px" }}
          />
          <Lable
            variant="body1"
            text={quantity.toString()}
            styles={{ justifyContent: "center" }}
          />
          <ButtonComp
            variant="outlined"
            startIcon={<RemoveIcon />}
            onClickHandler={removeItem}
            style={{ padding: "5px 1px 5px 10px" }}
          />
        </div>
      </div>
    </Card>
  );
};

export default FavouriteListItem;
