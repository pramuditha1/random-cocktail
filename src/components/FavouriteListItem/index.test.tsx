import React from "react";
import { render } from "@testing-library/react";
import FavouriteListItem, { FavouriteListItemPropTypes } from "./index";

test("FavouriteListItem matches snapshot", () => {
  const mockProps: FavouriteListItemPropTypes = {
    name: "test",
  quantity: 1,

  image: "https://www.thecocktaildb.com/images/media/drink/20d63k1504885263.jpg",
  addItem: jest.fn(),
  removeItem: jest.fn(),
  };

  const { asFragment } = render(
      <FavouriteListItem {...mockProps} />
  );

  expect(asFragment()).toMatchSnapshot();
});
