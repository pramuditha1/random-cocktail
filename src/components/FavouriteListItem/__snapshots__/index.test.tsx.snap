// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouriteListItem matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root flex items-center border border-gray-200 p-3 space-x-4  overflow-y-auto undefined css-13gi91s-MuiPaper-root-MuiCard-root"
  >
    <img
      alt="test-image"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img w-24 h-24 object-cover rounded css-o69gx8-MuiCardMedia-root"
      src="https://www.thecocktaildb.com/images/media/drink/20d63k1504885263.jpg"
    />
    <div
      class="flex flex-col items-center w-full space-y-1"
    >
      <div
        class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-2u1lx-MuiTypography-root"
      >
        test
      </div>
      <div
        class="flex items-center space-x-2"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSuccess MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSuccess css-1myamw5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AddIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-6uwl0p-MuiTypography-root"
        >
          1
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorError MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorError css-uaqvil-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="RemoveIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13H5v-2h14z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
